### PR TITLE
Restrict GET /users API to authenticated users

### LINKED ISSUE #72: Restrict GET /users API to authenticated users
## Description

Currently, for testing purposes, the GET /users API is open. Now that this is deployed this should be only accessible for authenticated users

### Update
- [x] Make GET /users restricted to authenticated users
- [x] Add a test to make sure unauthenticated users cannot access this API

## Definition of Done
- [x] All of the required items are completed.
- [x] PR should be sent to the `gsoc18-code` branch.
- [x] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by isabelcosta**
### Description

- [x] This PR restricts the GET /users API for authenticated users.
- [x] This should be merged after PR #64 which fixes some exception throwing from flask-jwt library.
- [x] Restrict some information of users, that should not be seen by other users. **(needs discussion)**
   - [x] Username -> visible
- [x] Make GET /users/verified also retricted

Fixes #72 

### Type of Change:

- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

- Tested GET /users API on Swagger UI
- Developed a test to attempt to access this API without an access token

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] Any dependent changes have been merged
- [x] Update Swagger documentation and the exported file at /docs folder

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by m-murad — APPROVED** [review]

**Comment by isabelcosta** [comment]
@m-murad @Dilu9218 until we add environment variables for the email verification on the deployed server, its best if we don't merge this, to better debug if a user is being registered into the system. To be able to check the users list without being logged in

**Comment by Dilu9218** [comment]
please solve conflicts in the same swagger.json :)

**Comment by isabelcosta** [comment]
@Dilu9218 conflicts are fixed!

**Review by Dilu9218 — APPROVED** [review]

**Pull request merged**
